packages/api/src/webhooks/cannon.ts

```diff
@@ -0,0 +1,253 @@
+import sql, { SQLStatement } from "sql-template-strings";
+import { DB } from "../store/db";
+import { Queue, Webhook, User, Stream } from "../schema/types";
+// import { getWebhooks } from "../controllers/helpers";
+import Model from "../store/model";
+import { fetchWithTimeout, fetchWithTimeoutAndSleep } from "../util";
+import logger from "../logger";
+import uuid from "uuid/v4";
+
+import { parse as parseUrl } from "url";
+import bytesToUuid from "uuid/lib/bytesToUuid";
+const isLocalIP = require("is-local-ip");
+const { Resolver } = require("dns").promises;
+// const resolver = new Resolver();
+
+const WEBHOOK_TIMEOUT = 5 * 1000;
+const MAX_BACKOFF = 10 * 60 * 1000;
+const BACKOFF_COEF = 1.2;
+
+export default class WebhookCannon {
+  db: DB;
+  store: Model;
+  running: boolean;
+  verifyUrls: boolean;
+  resolver: any;
```

Comment:
**nit:** You can fix the typing of this one by changing the `dns` require above to an `import`. I played around with this on the stream typescript thing, the pattern I liked was:
```ts
import { promises as dns } from "dns";
...
  resolver: dns.Resolver;
...
    this.resolver = new dns.Resolver();
```
WDYT?